RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT_vect.h

```diff
@@ -0,0 +1,318 @@
+#ifndef DAClusterizerInZT_vect_h
+#define DAClusterizerInZT_vect_h
+
+/**\class DAClusterizerInZT_vect
+
+ Description: separates event tracks into clusters along the beam line
+
+	Version which auto-vectorizes with gcc 4.6 or newer
+
+ */
+
+#include "RecoVertex/PrimaryVertexProducer/interface/TrackClusterizerInZ.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrack.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <vector>
+#include "DataFormats/Math/interface/Error.h"
+#include "RecoVertex/VertexTools/interface/VertexDistanceXY.h"
+#include "RecoVertex/VertexPrimitives/interface/TransientVertex.h"
+
+#include <memory>
+#include <stdlib.h>
+#include <malloc.h>
+
+template <typename T, std::size_t N = 16>
+class AlignmentAllocator {
+public:
+  typedef T value_type;
+  typedef std::size_t size_type;
+  typedef std::ptrdiff_t difference_type;
+
+  typedef T * pointer;
+  typedef const T * const_pointer;
+
+  typedef T & reference;
+  typedef const T & const_reference;
+
+  public:
+  inline AlignmentAllocator () throw () { }
+
+  template <typename T2>
+  inline AlignmentAllocator (const AlignmentAllocator<T2, N> &) throw () { }
+
+  inline ~AlignmentAllocator () throw () { }
+
+  inline pointer adress (reference r) {
+    return &r;
+  }
+
+  inline const_pointer adress (const_reference r) const {
+    return &r;
+  }
+
+  inline pointer allocate (size_type n) {
+    void * p = nullptr;
+    posix_memalign(&p,N,n*sizeof(value_type));
+    return (pointer)p;
+  }
+
+  inline void deallocate (pointer p, size_type) {
+    free(p);
+  }
+
+  inline void construct (pointer p, const value_type & wert) {
+     new (p) value_type (wert);
+  }
+
+  inline void destroy (pointer p) {
+    p->~value_type ();
+  }
+
+  inline size_type max_size () const throw () {
+    return size_type (-1) / sizeof (value_type);
+  }
+
+  template <typename T2>
+  struct rebind {
+    typedef AlignmentAllocator<T2, N> other;
+  };
+
+  bool operator!=(const AlignmentAllocator<T,N>& other) const  {
+    return !(*this == other);
+  }
+
+  // Returns true if and only if storage allocated from *this
+  // can be deallocated from other, and vice versa.
+  // Always returns true for stateless allocators.
+  bool operator==(const AlignmentAllocator<T,N>& other) const {
+    return true;
+  }
+};
+
+
+class DAClusterizerInZT_vect  final : public TrackClusterizerInZ {
+
+public:
+  template<typename T> 
+  using AlignedVector = std::vector<T, AlignmentAllocator<T, 16> >;
```

Comment:
all instances below are for T=double. So, one piece is 128 bytes, if I understood the alignment logic correctly.
The cache line is 64 bytes.
Is this choice optimal or did I miss something?